### PR TITLE
Todo example

### PR DESCRIPTION
**Pull request by kentan88**

### CODE REVIEW

File: modules/examples/src/todo/index.js [hunk 1]
```diff
@@ -70,6 +70,30 @@ class TodoApp {
   clearCompleted() {
     this.todoStore.removeBy((todo) => todo.completed);
   }
+
+  showAll() {
+    this.todoStore.list.forEach((todo) => {
+      todo.show = true;
```
Comment (vicb) [inline modules/examples/src/todo/index.js]: there is no `show` prop on `Todo`.
It should be declared & initialized for Dart

File: modules/examples/src/todo/index.js [hunk 2]
```diff
@@ -70,6 +70,30 @@ class TodoApp {
...
+    });
+  }
+
+  showActive() {
+    this.todoStore.list.forEach((todo) => {
```
Comment (vicb) [inline modules/examples/src/todo/index.js]: Use the `ListWrapper.forEach`  facade for interop Dart/JS

File: modules/examples/src/todo/index.js [hunk 3]
```diff
@@ -70,6 +70,30 @@ class TodoApp {
...
+      if(todo.completed)
+        todo.show = false;
```
Comment (vicb) [inline modules/examples/src/todo/index.js]: `todo.show = !todo.completed`
Factorize code with `showCompleted` ?

File: modules/examples/src/todo/services/TodoStore.js [hunk 4]
```diff
@@ -52,6 +52,10 @@ export class Store {
     this.spliceOut(record);
   }
 
+  save(record: KeyModel) {
```
Comment (vicb) [inline modules/examples/src/todo/services/TodoStore.js]: Why ? Looks useless

**Comment by vicb** [comment]
I've added general feedback. @davideast could probbaly tell if the changes make sense.

Please re-write your commit comments using the conventions in CONTRIBUTING.md

File: modules/examples/src/todo/todo.html [hunk 5]
```diff
@@ -51,13 +51,13 @@
     <div [class.hidden]="true"></div>
     <ul id="filters">
       <li>
-        <a href="#/" class="selected">All</a>
+        <a href="#/" class="selected" (click)="showAll()">All</a>
```
Comment (davideast) [inline modules/examples/src/todo/todo.html]: I think it is best to wait for the router before implementing the filter functionality. This way we can route based on the state of the todo list (all/active/completed).

File: modules/examples/src/todo/index.js [hunk 6]
```diff
@@ -70,6 +70,30 @@ class TodoApp {
...
+      else
+        todo.show = true;
+    });
+  }
+
+  showCompleted() {
+    this.todoStore.list.forEach((todo) => {
```
Comment (davideast) [inline modules/examples/src/todo/index.js]: This is another example of where routing will help determine the contents of the list.

File: modules/examples/src/todo/todo.html (hunk 5 above)
Comment (vsavkin) [inline modules/examples/src/todo/todo.html]: +1

**Comment by naomiblack** [comment]
@vicb per Misko's label, looks ready to go? Can you check?

**Comment by vicb** [comment]
@kentan88 could you please take the feedback into account, rebase on master & squash the commits. Do not hesitate to ask if you need help. Thanks.

**Comment by vicb** [comment]
@kentan88 closing this PR as it doesn't get updated. Feel free to re-open if you have time to take the feedback into account & update. Thanks.

**Pull request closed without merging**